renderer/components/main-content/index.tsx

```diff
@@ -163,41 +163,107 @@ const MainContent = ({
+        ) {
+          const reader = new FileReader();
+          reader.onload = async (event) => {
+            const result = event.target?.result;
+            if (typeof result === "string") {
+              file.encodedBuffer = Buffer.from(result, "utf-8").toString(
+                "base64",
+              );
+            } else if (result instanceof ArrayBuffer) {
+              file.encodedBuffer = Buffer.from(new Uint8Array(result)).toString(
+                "base64",
+              );
+            } else {
+              logit("🚫 Invalid file pasted");
+              toast({
+                title: t("ERRORS.INVALID_IMAGE_ERROR.TITLE"),
+                description: t(
+                  "ERRORS.INVALID_IMAGE_ERROR.CLIPBOARD_DESCRIPTION",
+                ),
+              });
+            }
+            window.electron.send(ELECTRON_COMMANDS.PASTE_IMAGE, file);
+          };
+          reader.readAsArrayBuffer(fileObject);
+        } else {
+          logit("🚫 Invalid file pasted");
+          toast({
+            title: t("ERRORS.INVALID_IMAGE_ERROR.TITLE"),
+            description: t("ERRORS.INVALID_IMAGE_ERROR.CLIPBOARD_DESCRIPTION"),
+          });
+        }
+      } else {
+        logit("🚫 Invalid file pasted");
+        toast({
+          title: t("ERRORS.INVALID_IMAGE_ERROR.TITLE"),
+          description: t("ERRORS.INVALID_IMAGE_ERROR.CLIPBOARD_DESCRIPTION"),
+        });
+      }
+    } else {
       toast({
-        title: t("ERRORS.INVALID_IMAGE_ERROR.TITLE"),
-        description: t("ERRORS.INVALID_IMAGE_ERROR.ADDITIONAL_DESCRIPTION"),
+        title: t("ERRORS.NO_OUTPUT_FOLDER_ERROR.TITLE"),
+        description: t("ERRORS.NO_OUTPUT_FOLDER_ERROR.DESCRIPTION"),
       });
-      return;
     }
-    const type = items[0].type;
-    const filePath = files[0].path;
-    const extension = files[0].name
-      .split(".")
-      .at(-1)
-      .toLowerCase() as ImageFormat;
-    logit("📋 Pasted file: ", JSON.stringify({ type, filePath, extension }));
-    if (!type.includes("image") && !VALID_IMAGE_FORMATS.includes(extension)) {
+  };
+
+  useEffect(() => {
+    // Events
+    const handlePasteEvent = (e) => handlePaste(e);
+    const handlePasteImageSaveSuccess = (_: any, imageFilePath: string) => {
+      setImagePath(imageFilePath);
+      validateImagePath(imageFilePath);
+    };
+    const handlePasteImageSaveError = (_: any, error: string) => {
       toast({
-        title: t("ERRORS.INVALID_IMAGE_ERROR.TITLE"),
-        description: t("ERRORS.INVALID_IMAGE_ERROR.ADDITIONAL_DESCRIPTION"),
+        title: t("ERRORS.NO_IMAGE_ERROR.TITLE"),
+        description: error,
       });
-    } else {
-      setImagePath(filePath);
-      const dirname = getDirectoryFromPath(filePath);
-      logit("🗂 Setting output path: ", dirname);
-      if (!FEATURE_FLAGS.APP_STORE_BUILD) {
-        if (!rememberOutputFolder) {
-          setOutputPath(dirname);
-        }
-      }
-      validateImagePath(filePath);
```

Comment:
Am I reading this wrong, or does this remove the existing functionality of processing pasted file paths?

Reply:
It has been moved under the useEffect hook's handleOnPasteSuccess function..